src/components/state-details.js

```diff
@@ -78,87 +22,72 @@ const slugToTitle = (placeName) => {
   return words.join(' ')
 }
 
-const getPlacesData = (data) => {
-  const allPlacesData = {}
-  places.forEach((place, i) => {
-    const placeTitle = slugToTitle(place)
-    allPlacesData[place] = {}
-    allPlacesData[place]['name'] = placeTitle
-
-    // grab the state total employed from totals
-    allPlacesData[place]['emissions'] = data[place]
-  })
-  return allPlacesData
-}
-
 const currentYear = new Date().getFullYear()
-
 // We want to get to 0 by 2050 and we use our current emissions as a start,
 // so the % to cut by is 100 divided by the number of years we have
 const cutPerYearPrcnt = (100 / (2050 - currentYear)).toFixed(1)
 
 const StateDetailsPage = ({location, data}) => {
+  // place info and string
   const currentPlace = location.pathname.split("/")[1]
-  const placesData = getPlacesData(data.emissionsJson)
-  const currPlaceData = placesData[currentPlace]
+  // clean up title as needed
+  const placeTitle = slugToTitle(currentPlace)
+  const stateFaceClass = currentPlace.toLowerCase().replaceAll(' ', '-')
+
+  // Each json loads in as an allSomethingJson and is filtered for 
+  // data relevant to this state, which is great!
+  // the first edge node will have the relevant data, 
+  // so we can just take the first index
+
+  // #### EMISSIONS #### 
+  const emissionsByYear = data.allEmissionsJson.edges[0].node.emissionsByYear
+  const latestEmissions = emissionsByYear[emissionsByYear.length - 1]
+  // desstructure out the different emissions categories for simplicity below
+  const {
+    buildings: buildingsEmissions,
+    dirty_power: dirtyPowerEmissions,
+    dumps_farms_industrial_other: farmsDumpsOtherEmissions,
+    transportation: transportionEmissions
+  } = latestEmissions
+  
+  // sum, then make nice percentages
+  const sumOfEmissions = buildingsEmissions + dirtyPowerEmissions + farmsDumpsOtherEmissions + transportionEmissions
+  const buildingsPrcnt = (buildingsEmissions / sumOfEmissions * 100).toFixed(0)
+  const powerPrcnt = (dirtyPowerEmissions / sumOfEmissions * 100).toFixed(0)
+  const transportPrcnt = (transportionEmissions / sumOfEmissions * 100).toFixed(0)
+  const otherPrcnt = (farmsDumpsOtherEmissions / sumOfEmissions * 100).toFixed(0)
 
+  
+  // #### VEHICLES #### 
   const {
     Cars_All: carsAll,
     EV_Registration: evRegistration,
-  } = data.vehiclesJson[currentPlace]
+  } = data.allVehiclesJson.edges[0].node
```

Comment:
This is the new proposed pattern for accessing data, the `edges[0].node` syntax is fairly clunky, but as far as I can tell just a fact of life for graphql!